### DISCUSSION TITLE
rebuild Utl

### DISCUSSION THREAD

**Discussion by developmentd3** [nftchef/art-engine#62] [Help + Q&A] [answered]
Love to use this, but it doesn't seem to be taking any arguments, and editing the _dna.json file before running it, doesn't seem to effect the output?

Anything I could be missing?

P.s. absolutely amazing fork!!

**Comment by developmentd3** [comment, marked as answer]
well I'd like your input, but I know what I did manually, is i focused on the layer name in the _dna file instead of the index numering. I.e
0.0:NONE.png 
is not the same as 
0.1:NONE.png

Thanks again!!

**Comment by nftchef** [comment]
Hi! If you're looking at the rebuild all util, you can get a list of the expected command syntax with --help

```bash
 node utils/rebuildAll.js --help

Usage: rebuildAll [options]

Options:
  -o, --omit <omitIndex>         omit any given layer by layer index
  -i, --startIndex <startIndex>  Then num.to start Output naming at, default is
                                 0
  -b, --background <background>  override the config generate background bool
  -s, --source <source>          Optional source path of _dna.json
  -d, --debug                    display additional logging
  -v, --verbose                  display even more additional logging
  -h, --help                     display help for command
  ```
  
  and the numbering 0:0 is a stored "map" to folder indices in the layers Directory. If the layers folder changes, then the DNA will be invalid.  for example, if you change a filename, you would need to find and replace all occurrences of the original name in _dna.json for the script to work (and other scripts that rely on _dna.json)
